queenbee/schema/workflow.py

```diff
@@ -4,39 +4,63 @@
 end steps for the workflow.
 """
 from queenbee.schema.qutil import BaseModel
-from pydantic import Schema, validator
+from pydantic import Field, validator, constr, root_validator
 from typing import List, Union
-from enum import Enum
 from uuid import UUID, uuid4
 import json
 import os
 from queenbee.schema.dag import DAG
 from queenbee.schema.arguments import Arguments
 from queenbee.schema.operator import Operator
 from queenbee.schema.function import Function
+from queenbee.schema.artifact_location import LocalLocation, HTTPLocation, S3Location
 
 
 class Workflow(BaseModel):
     """A DAG Workflow."""
-    type: Enum('Workflow', {'type': 'Workflow'}) = 'Workflow'
+
+    type: constr(regex='^workflow$')
```

Comment:
Was it always there and we didn't know? This is much cleaner.

Reply:
Yep... :smile: